driver-dse/src/main/java/com/datastax/driver/graph/GraphTraversalResult.java

```diff
@@ -0,0 +1,100 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.graph;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.exceptions.DriverException;
+
+public class GraphTraversalResult {
+    private final String jsonString;
+
+    private JsonNode rootNode;
+    private ObjectMapper objectMapper;
+
+    private Row row;
+
+    GraphTraversalResult(Row row, ObjectMapper objectMapper) {
+        this.jsonString = row.getString("gremlin");
+        this.row = row;
+        this.objectMapper = objectMapper;
+        try {
+            this.rootNode = this.objectMapper.readTree(this.jsonString).get("result");
+        } catch (IOException e) {
+            throw new DriverException("Could not parse the result returned by the Graph server as a JSON string : " + this.jsonString);
+        }
+    }
+
+    static GraphTraversalResult fromRow(Row row, ObjectMapper objectMapper) {
+        return row == null ? null : new GraphTraversalResult(row, objectMapper);
+    }
+
+    /**
+     * API
+     */
+
+    /**
+     * Get the raw JSON string returned by the server.
+     *
+     * @return the raw JSON string received from the server.
+     */
+    public String getResultString() {
+        return this.jsonString;
+    }
+
+    /**
+     * Get the {@link com.datastax.driver.graph.GraphData} object for the specified key.
+     *
+     * @param keyOrIndex Can be either a String or a int/Integer. Must not be null, or an exception will be thrown.
+     * @return A GraphData instance containing the value for the Json entity required.
+     */
+    public GraphData get(Object keyOrIndex) {
```

Comment:
what do you think about also adding a `has` method?  Since get delegates to rootNode.get(x) which throws an Exception when no key/index exists, a `has` would be a nice check-before construct.